src/composables/sdk.ts

```diff
@@ -26,10 +29,29 @@ export function useSdk({ store }: IDefaultComposableOptions) {
   const isNodeConnecting = computed(() => nodeStatus.value === NODE_STATUS_CONNECTING);
   const isNodeReady = computed(() => nodeStatus.value === NODE_STATUS_CONNECTED);
   const isNodeError = computed(() => nodeStatus.value === NODE_STATUS_ERROR);
-  const isNodeMainnet = computed(() => activeNetwork.value.networkId === NETWORK_ID_MAINNET);
-  const isNodeTestnet = computed(() => activeNetwork.value.networkId === NETWORK_ID_TESTNET);
+
+  const isNodeMainnet = computed(() => nodeNetworkId.value === NETWORK_ID_MAINNET);
+  const isNodeTestnet = computed(() => nodeNetworkId.value === NETWORK_ID_TESTNET);
   const isCustomNodeNetwork = computed(() => !isNodeMainnet.value && !isNodeTestnet.value);
 
+  const isTippingSupported = computed(() => (RUNNING_IN_TESTS && !isCustomNodeNetwork.value));
+
+  const dexContracts = computed(
+    () => nodeNetworkId.value ? DEX_CONTRACTS[nodeNetworkId.value] : undefined,
+  );
+
+  async function createNewNodeInstance(url: string): Promise<Node | null> {
+    try {
+      const nodeInstance = await Node({ url });
+      nodeNetworkId.value = nodeInstance.nodeNetworkId;
+      return nodeInstance;
+    } catch (error) {
+      nodeNetworkId.value = undefined;
+      store.commit('setNodeStatus', NODE_STATUS_ERROR);
+      throw new Error();
+    }
+  }
```

Comment:
Yes, you have totally right. But please bare in mind this is the old SDK composable, so here we are doing some not fully beautiful magic to make stuff working and prepare the app for fully moving to SDK13.

With SDK13 Node instance creation is synchronous, we don't need to wait for it. So you are right - only way to establish if we are connected is by running the `getStatus()` method, which is asynchronous and returns the `networkId`. We will definitely switch to this approach. I'll add a code comment about this.

For now as the previous version of the Node instantiation is asynchronous I'd like to avoid additional calls as the app is already heavy on them. The Node instantiation is returning the `networkId` so why not using this for now? For me it's reasonable.

> It is updating a global variable nodeNetworkId which has a pretty big significant effect across this project.
Yes, exactly. This task is about this major refactor. I already mentioned this in our chat but let me explain this again.

Previously when user added new custom network the code we had was setting the `networkId` value to `ae_mainnet`. So no matter what URL you were using our app was treating those networks as the **Mainnet** :O This is bad, yes?

Recently I changed this and all custom networks has the `networkId` set to `"custom"`. But this is bad, because how to write logic, that would check if user is connected with the mainnet? Let's say that user wants to use the mainnet url but only change the MDW, so he will add new network, leave the node URL but provide custom mdw URL. He will assume that he is using the mainnet, but our app would treat it as something different and would hide some features that are inclusive only for the mainnet. This is albo bad, yes?

So this is why we are listening to what the node is telling us and assuming that this is kind of true. User will still see the network name as something custom, but our logic can display/hide some features based on this.

What do you think?

Reply:
Okay, only because of the SDK migration story it may make sense to keep this side effect.

> For the mainnet we are displaying "Buy" button and dashboard card, but on Testnet it's the faucet. On totally custom networks there is no button.

I like this idea but there is one more thing, how are you planning to handle currency conversions? (or are you planning to disable the FIAT conversions)

In the galaxy event, I heard hyperchain network creators could create exchange or currency conversion for their network too. Hence we can add one more optional textbox input in the Network Form to supply currency conversion backend URL.

Reply:
Regarding the currency conversions there is a separate task for this: https://aeternity-blockchain.atlassian.net/browse/SW-894
And for now the custom networks won't have the currencies displayed at the beginning. Next step will be to add the "Coin" details info to the status endpoint of the Node. So instead of adding the AE coin by our code we will be receiving it from the Node.

Reply:
Got it. Including the coin logo as well.